client/state/sites/selectors.js

```diff
@@ -985,3 +985,19 @@ export function getCustomizerUrl( state, siteId ) {
 		'return': returnUrl
 	}, adminUrl );
 }
+
+/**
+ * Determines if site's permalinks are editable
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} siteId Site ID
+ * @return {Boolean} true if the site's permalinks are editable
+ */
+export function areSitePermalinksEditable( state, siteId ) {
+	const site = getRawSite( state, siteId );
```

Comment:
Would be good to leverage [`getSiteOption`](https://github.com/Automattic/wp-calypso/blob/d629ffc335fb509a5964766001441f3ed37447e4/client/state/sites/selectors.js#L308-L323) here I think.